internal/k8shandler/logstore.go

```diff
@@ -516,3 +524,180 @@ func isNodeDifferent(current, desired elasticsearch.ElasticsearchNode) (elastics
 
 	return current, different
 }
+
+func (clusterRequest *ClusterLoggingRequest) updateLokiCRIfRequired(desired *lokiv1beta1.LokiStack) (err error) {
+	current := &lokiv1beta1.LokiStack{}
+
+	if err = clusterRequest.Get(desired.Name, current); err != nil {
+		if apierrors.IsNotFound(err) {
+			// the object doesn't exist yet
+			// recreate it on the next time
+			return nil
+		}
+		return fmt.Errorf("failed to get LokiStack CR: %v", err)
+	}
+
+	different := false
+	if current.Spec.Size != desired.Spec.Size {
+		log.Info("LokiStack size policy change found, updating", "currentName", current.Name)
+		current.Spec.Size = desired.Spec.Size
+		different = true
+	}
+
+	if current.Spec.StorageClassName != desired.Spec.StorageClassName {
+		log.Info("LokiStack storage class name policy change found, updating", "currentName", current.Name)
+		current.Spec.StorageClassName = desired.Spec.StorageClassName
+		different = true
+	}
+
+	if !reflect.DeepEqual(current.Spec.Storage, desired.Spec.Storage) {
+		log.Info("LokiStack object storage spec change found, updating", "currentName", current.Name)
+		current.Spec.Storage = desired.Spec.Storage
+		different = true
+	}
+
+	if !reflect.DeepEqual(current.Spec.Template, desired.Spec.Template) {
+		log.Info("LokiStack template spec change found, updating", "currentName", current.Name)
+		current.Spec.Template = desired.Spec.Template
+		different = true
+	}
+
+	if different {
+		if err = clusterRequest.Update(current); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cr *ClusterLoggingRequest) emptyLokiCR() *lokiv1beta1.LokiStack {
+	return &lokiv1beta1.LokiStack{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      lokiResourceName,
+			Namespace: cr.Cluster.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "LokiStack",
+			APIVersion: lokiv1beta1.GroupVersion.String(),
+		},
+		Spec: lokiv1beta1.LokiStackSpec{},
+	}
+}
+
+func (cr *ClusterLoggingRequest) newLokiCR() *lokiv1beta1.LokiStack {
+
+	logStoreSpec := logging.LogStoreSpec{}
+	if cr.Cluster.Spec.LogStore != nil {
+		logStoreSpec = *cr.Cluster.Spec.LogStore
+	}
+
+	var (
+		size              lokiv1beta1.LokiStackSizeType
+		replicationFactor int32
+		template          lokiv1beta1.LokiTemplateSpec
+	)
+	switch logStoreSpec.LokiStackSpec.Size {
+	case logging.SizeOneXMedium:
+		size = lokiv1beta1.SizeOneXMedium
+		replicationFactor = 3
+	default:
+		size = lokiv1beta1.SizeOneXSmall
+		replicationFactor = 2
+	}
```

Comment:
Can you elaborate on why we declare statically a replication factor at all? Isn't this obsolete too? AFAICS the loki-operator will declare one from the per defaults per t-shirt size ([1x.small](https://github.com/grafana/loki/blob/main/operator/internal/manifests/internal/sizes.go#L235), [1x.medium](https://github.com/grafana/loki/blob/main/operator/internal/manifests/internal/sizes.go#L284))

Reply:
it's set to `required` and minimum 1 in our lokiStack api https://github.com/grafana/loki/blob/main/operator/api/v1beta1/lokistack_types.go#L511-L517